### PR TITLE
TCM-448 | Implement skeleton loading in `Discover`

### PR DESCRIPTION
**Pull request by VivekV95**
## Description
Display skeletons in `Discover` when videos are loading

### Summary

- [x] Bring over skeleton implementations from Xpass
- [x] Display skeletons in Discover
- [x] Display either all workouts or filtered workouts depending on the existence of selected filters
- [x] Reset filters temporarily only when `Programs` or `Collections` is selected
- [x] Remove some unnecessary code and files  

### Issue

[TCM-448](https://hyfnla.atlassian.net/browse/TCM-448)

### Video/Screenshot

[With selected filters](https://drive.google.com/file/d/172Z2Ycdk-a_pNY9yY9PYy51SLCtpW9JW/view?usp=sharing)
[No selected filters](https://drive.google.com/file/d/1q0XW2CgaTT7edY_jA0SAQFuzbr-HEkqe/view?usp=sharing)
[Clear filters](https://drive.google.com/file/d/1Uu1yIdOA2E8rsQqPk4PfoBWLedqvr6N_/view?usp=sharing)

### CODE REVIEW

File: app/src/main/java/com/trx/consumer/screens/discover/DiscoverViewModel.kt [hunk 1]
```diff
@@ -5,33 +5,31 @@ import androidx.lifecycle.viewModelScope
...
+) : BaseViewModel(), VideoWorkoutListener, DiscoverFilterListener {
 
     //region Objects
-    var workouts: List<VideoModel> = listOf()
-    var collections: List<VideosModel> = listOf()
-    var programs: List<VideosModel> = listOf()
+
+    val model = DiscoverModel.skeleton()
     var params: FilterParamsModel = FilterParamsModel()
     var filters: List<FilterModel> = listOf()
```
Comment (onodude) [inline app/src/main/java/com/trx/consumer/screens/discover/DiscoverViewModel.kt:27]: do we need the filters variable can we use DiscoverModel.filters ? (we could rename lstFilters, filters too)

Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/screens/discover/DiscoverViewModel.kt:27]: Can put it inside `DiscoverModel`

**Review by onodude — CHANGES_REQUESTED** [review]
thanks @VivekV95 added some comments

File: app/src/main/java/com/trx/consumer/screens/discover/DiscoverViewModel.kt (hunk 1 above)
Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/screens/discover/DiscoverViewModel.kt]: [Done](https://github.com/hyfn/trx-consumer-android/pull/98/files#diff-ae458370b56efa34eba0c2e6d926871fd4f8ce3dd6d190fdafeb0a586ef57838R10)

**Review by onodude — APPROVED** [review]
thank you!

**Pull request merged**